internal/service/directconnect/connection_test.go

```diff
@@ -221,6 +259,30 @@ resource "aws_dx_connection" "test" {
 `, rName)
 }
 
+func testAccConnectionConfig_macsecEnabled(rName string) string {
+	return fmt.Sprintf(`
+data "aws_dx_locations" "test" {}
+
+locals {
+  location_codes = tolist(data.aws_dx_locations.test.location_codes)
+  idx            = min(2, length(local.location_codes) - 1)
+}
+
+data "aws_dx_location" "test" {
+  location_code = local.location_codes[local.idx]
+}
+
+resource "aws_dx_connection" "test" {
+  name             = %[1]q
+  bandwidth        = "100Gbps"
+  location         = data.aws_dx_location.test.location_code
+  macsec_requested = true
```

Comment:
The API uses a parameter name of `requestMACSec`. Consider using `request_macsec` as the field name for the schema as well.
```suggestion
  request_macsec = true
```